.circleci/config.yml

```diff
@@ -161,6 +161,75 @@ orbs:
               key: bundle-{{ .Environment.CIRCLE_CACHE_VERSION }}-<<parameters.ruby_version>>-{{ checksum "lib/ddtrace/version.rb" }}-{{ .Branch }}-{{ checksum "Gemfile" }}-{{ checksum "Appraisals" }}-{{ checksum "ddtrace.gemspec" }}-{{ checksum ".circleci/bundle_checksum" }}'
               paths:
                 - /usr/local/bundle
+      build_integration:
+        executor: docker-integration
+        parameters:
+          ruby_version:
+            description: Ruby version
+            type: string
+          integration_apps:
+            description: Integration apps
+            type: string
+        steps:
+          - checkout
+          - setup_remote_docker:
+              docker_layer_caching: true
+          - run:
+              name: Build integration base images
+              command: ./integration/bin/build-images -v <<parameters.ruby_version>>
+          - run:
+              name: Build integration app images
+              command: |
+                for i in <<parameters.integration_apps>>; do
+                  ./integration/apps/$i/bin/build-images -v <<parameters.ruby_version>>
+                done
+      test_integration:
+        executor: docker-integration
+        parameters:
+          ruby_version:
+            description: Ruby version
+            type: string
+          integration_apps:
+            description: Integration apps
+            type: string
+        steps:
+          - checkout
+          - setup_remote_docker:
+              docker_layer_caching: true
+          - run:
+              name: Test integration apps
+              command: |
+                for i in <<parameters.integration_apps>>; do
+                  ./integration/apps/$i/bin/ci -v <<parameters.ruby_version>>
+                done
+      build_and_test_integration:
```

Comment:
Ah, I think this is an artifact that I forgot to remove.

There's an advantage to building then testing in that you don't have to rebuild some images multiple times (e.g. `wrk`, `agent`) and that if testing fails (say because of a timeout), then you can restart from the test step rather than the build step. So this is all about cutting down on build times.

In practice, it seems like `docker_layer_caching` is not consistent in propagating the built images between steps; something we could fix by implementing Circle's caching step ourselves. However, this takes time to setup and test to get it right, and is an optimization that isn't necessary for V1. Maybe something we need in V2 if we find build times are too painful.

So for now, I'll remove the separate build & test steps, and use the combined one instead.